src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
+to use synchronous loggers,
+* or
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/BasicAsyncLoggerContextSelector.html[`org.apache.logging.log4j.core.async.BasicAsyncLoggerContextSelector`]
+to use asynchronous loggers.
+
+In this approach you must use xref:manual/lookups.adoc[lookups] to register the application that generated a log event.
+The most useful lookups in this case are:
+
+Web lookup::
+It does not require any setup, but it is available only after `Log4jServletFilter` has been executed.
+Some log events pertinent to a web application can be unmarked.
+See xref:manual/lookups.adoc#WebLookup[web lookup] for more information.
+
+JNDI lookup::
+It covers a larger part of the handling of a request, but it requires additional setup to export the name of the application via JNDI.
+See xref:manual/lookups.adoc#JndiLookup[JNDI lookup] for more information.
+
+When using a single logger context you choose between:
+
+* logging all events to a single appender.
+You'll need a tool like Elastic Search to separate log events.
+We recommend the usage of a structured layout like
+xref:manual/json-template-layout.adoc[]
+in this case:
++
+[tabs]
+====
+XML::
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/log4j2-single.xml[tag=global]
 ----
 
-This can be slightly more convenient when using Java 1.8 and lambda functions as demonstrated below.
+JSON::
++
+[source,json,indent=0]
+----
+include::example$manual/webapp/log4j2-single.json[tag=global]
+----
 
-[source,java]
+YAML::
++
+[source,yaml,indent=0]
+----
+include::example$manual/webapp/log4j2-single.yaml[tag=global]
 ----
-import java.io.IOException;
-import javax.servlet.AsyncContext;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.apache.logging.log4j.web.WebLoggerContextUtils;
+Properties::
++
+[source,properties,indent=0]
+----
+include::example$manual/webapp/log4j2-single.properties[tag=global]
+----
+====
 
-public class TestAsyncServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        final AsyncContext asyncContext = req.startAsync();
-        asyncContext.start(WebLoggerContextUtils.wrapExecutionContext(this.getServletContext(), () -> {
-            final Logger logger = LogManager.getLogger(TestAsyncServlet.class);
-            logger.info("Hello, servlet!");
-        }));
-    }
-}
+* logging events to a separate appender for each application.
+In this case you can use
+xref:manual/appenders.adoc#RoutingAppender[routing appender]
+to separate the events:
++
+[tabs]
+====
+XML::
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/log4j2-single.xml[tag=routing]
```

Comment:
I am still in favor of using JTL – that is what we recommend for production. (You can leave the `FileAppender` usage there.)